autofs-5.0.9 - check for non existent negative entries in lookup_ghost()

From: Ian Kent <raven@themaw.net>

Map entries that have been created in the cache due to a negative lookup
but don't exist in the map source shouldn't have directories created.
In fact map entries that are negative shouldn't have directories created
at all since they should have neen created befor the entry became negative.
---
 CHANGELOG       |    1 +
 daemon/lookup.c |    8 ++++++++
 2 files changed, 9 insertions(+)

--- autofs-5.0.5.orig/CHANGELOG
+++ autofs-5.0.5/CHANGELOG
@@ -150,6 +150,7 @@
 - extend fix for crash due to thread unsafe use of libldap.
 - check scandir() return value.
 - fix use cache entry after free in lookup_prune_one_cache().
+- check for non existent negative entries in lookup_ghost().
 
 03/09/2009 autofs-5.0.5
 -----------------------
--- autofs-5.0.5.orig/daemon/lookup.c
+++ autofs-5.0.5/daemon/lookup.c
@@ -594,6 +594,14 @@ int lookup_ghost(struct autofs_point *ap
 		cache_readlock(mc);
 		me = cache_enumerate(mc, NULL);
 		while (me) {
+			/*
+			 * Map entries that have been created in the cache
+			 * due to a negative lookup shouldn't have directories
+			 * created if they haven't already been created.
+			 */
+			if (!me->mapent)
+				goto next;
+
 			if (!strcmp(me->key, "*"))
 				goto next;
 
